apps/src/lab2/projects/projectsApi.ts

```diff
@@ -7,13 +7,17 @@ const rootUrl = '/projects/';
 // get the project identifier (channel id) for that level (and script, if provided).
 export async function getChannelForLevel(
   levelId: number,
-  scriptId?: number
+  scriptId?: number,
+  userId?: string
 ): Promise<Response> {
   let requestString = rootUrl;
   if (scriptId !== undefined) {
     requestString += `script/${scriptId}/`;
   }
   requestString += `level/${levelId}`;
+  if (userId !== '') {
+    requestString += `/user/${userId}`;
+  }
```

Comment:
This can be called where `userId` is undefined, right? (the param is optional)

Should it use the same `userId !== undefined` check as the `scriptId` one?

Reply:
Good point.  It wasn't really utilizing the optional part before, but it does now.